app/services/post_alerter.rb

```diff
@@ -265,6 +269,31 @@ def category_watchers(topic)
       .pluck(:user_id)
   end
 
+  def category_or_tag_muters(topic)
+    User
+      .joins(
+        "LEFT JOIN category_users ON users.id = category_users.user_id AND #{
+          DB.sql_fragment(
+            "category_users.category_id = :category_id AND category_users.notification_level = :muted",
+            category_id: topic.category_id,
+            muted: CategoryUser.notification_levels[:muted],
+          )
+        }",
+      )
+      .joins(
+        "#{DB.sql_fragment("LEFT JOIN topic_tags ON topic_tags.topic_id = :topic_id", topic_id: topic.id)}",
+      )
+      .joins(
+        "LEFT JOIN tag_users ON users.id = tag_users.user_id AND #{
+          DB.sql_fragment(
+            "tag_users.tag_id IN (tag_users.tag_id) AND tag_users.notification_level = :muted",
+            muted: TagUser.notification_levels[:muted],
+          )
+        }",
+      )
+      .where("category_users.id IS NOT NULL OR tag_users.id IS NOT NULL")
```

Comment:
```suggestion
    User
      .joins(
        "LEFT JOIN category_users ON users.id = category_users.user_id AND category_users.category_id = #{topic.category_id.to_i} AND category_users.notification_level = #{CategoryUser.notification_levels[:muted].to_i}"
      )
      .joins(
        "LEFT JOIN topic_tags ON topic_tags.topic_id = #{topic.id.to_i}"
      )
      .joins(
        "LEFT JOIN tag_users ON users.id = tag_users.user_id AND tag_users.tag_id IN (tag_users.tag_id) AND tag_users.notification_level = #{TagUser.notification_levels[:muted].to_i}"
      )
      .where("category_users.id IS NOT NULL OR tag_users.id IS NOT NULL")
```

I feel like this is much easier read and just as safe as compared to using `DB.sql_fragment.`